crypto-jni/src/lib.rs

```diff
@@ -179,28 +182,83 @@ pub extern "system" fn Java_com_concordium_sdk_crypto_CryptoJniNative_generatePu
 #[derive(SerdeSerialize, SerdeDeserialize)]
 enum CryptoJniResult<T> {
     Ok(T),
-    Err(jint),
+    Err(JNIError),
+}
+
+#[derive(SerdeSerialize, SerdeDeserialize)]
+#[allow(non_snake_case)]
+enum JNIErrorType {
```

Comment:
So this is more of a status code than an actual `Error` right? 

I suggest renaming it to something that differentiates it from "normal" rust errors.